src/Polysemy/Error.hs

```diff
@@ -242,7 +243,7 @@ mapError f = interpretH $ \case
 {-# INLINE mapError #-}
 
 
-newtype WrappedExc e = WrappedExc { unwrapExc :: e }
+data WrappedExc e = WrappedExc { getExcUid :: {-# UNPACK #-} !Unique, unwrapExc :: e }
   deriving (Typeable)
 
 instance Typeable e => Show (WrappedExc e) where
```

Comment:
When the type argument is dropped, the `Show` instance should be replaced. The very wordy instance from in-other-words is probably warranted (adapted to fit polysemy, of course).